Update cigarette log EMA logic and view.

eschlange opened the pull request:
[#95522106]

ericcf commented on src/partials/cigarette_log.html:
```diff
@@ -1,20 +1,41 @@
         </button>
         <br>
-        <button ng-click="smoked = false && cigarette.saveCigaretteLog()" ng-show="smoked == null" class="btn btn-lg btn-info" tabindex="0" aria-hidden="false">
+        <button ng-click="cigarette.cigarette_log[cigarette.CURRENTLY_SMOKING] = false" ng-show="cigarette.cigarette_log[cigarette.CURRENTLY_SMOKING] === undefined" class="btn btn-lg btn-info" tabindex="1">
```
I'd like to see some of the logic on this page moved to the controller. Unless you're writing e2e or appium tests for this, I'm not sure how you'll prevent regressions.

eschlange commented:
@ericcf Hey Eric, I refactored the logic out of the view and added specs. Please let me know what you think. I'll need to manually merge after you give the go-ahead.

ericcf commented:
This looks way cleaner to me. Thanks!
